pkg/proxy/iptables/proxier.go

```diff
@@ -1376,7 +1391,6 @@ func (proxier *Proxier) syncProxyRules() {
 
 			// Create the endpoint chain
 			natChains.Write(utiliptables.MakeChainLine(endpointChain))
-			activeNATChains[endpointChain] = true
```

Comment:
Oh... so, the reason the earlier loop is there is because we need to make sure `activeNATChains` has all the right elements even if we do a partial sync. Originally, doing a partial sync meant doing `continue` and skipping over the entire second half of the loop, but that got changed, with the whole `skippedNatChains` thing, to fix the metrics... so yeah, I could simplify this.